include/envoy/network/transport_socket.h

```diff
@@ -233,6 +233,11 @@ class TransportSocketFactory {
    * are already loaded.
    */
   virtual void addReadyCb(std::function<void()> callback) PURE;
+
+  /**
+   * Check whether matched transport socket which required to use secret information is available.
+   */
+  virtual bool secureTransportReady() const PURE;
```

Comment:
I would just name this `isReady`.

Reply:
It will be always false when raw buffer socket. So I prefer `secureTransportReady`.

Reply:
actually that would be oppose, raw buffer socket should always return `true` for this. It will be consistent with `addReadyCb` behavior above.

Reply:
done